src/lib/workspace/Variable.tsx

```diff
@@ -63,3 +76,19 @@ export function VariableDetails(props: Props) {
     </ErrorBoundary>
   );
 }
+
+function findMultifilterVariableLeaves(
+  variable: MultiFilterVariable,
+  variablesByParentId: Record<string, VariableTreeNode[] | undefined>
+): Variable[] {
+  if (variable.parentId == null) return [];
+  const variables = variablesByParentId[variable.id] ?? [];
+  for (const child of variables) {
+    for (const grandchild of variablesByParentId[child.id] ?? []) {
+      if (grandchild) variables.push(grandchild);
```

Comment:
I'm not familiar with the layout of the data structure, so I'll take your word for it that we only want grandchildren (and not the children).

Reply:
The variable name `grandchild` is not really accurate. What this loop is doing is: for each variable, find its children (those whose `parentId` is the same as the variable's `id`), and add it to the array `variables`. As the array is being iterated, elements that are `push`ed to it will be included in the iteration.